datalad/interface/utils.py

```diff
@@ -241,289 +218,12 @@ def _fx(res, **kwargs):
-                # loops, i.e. when a hook yields a result that
-                # triggers that same hook again
-                for hr in run_jsonhook(hook, spec, r, dataset_arg):
-                    # apply same logic as for main results, otherwise
-                    # any filters would only tackle the primary results
-                    # and a mixture of return values could happen
-                    if not keep_result(hr, result_filter, **allkwargs):
-                        continue
-                    hr = xfm_result(hr, result_xfm)
-                    # rationale for conditional is a few lines down
-                    if hr:
-                        yield hr
-        if not keep_result(r, result_filter, **allkwargs):
-            continue
-        r = xfm_result(r, result_xfm)
-        # in case the result_xfm decided to not give us anything
-        # exclude it from the results. There is no particular reason
-        # to do so other than that it was established behavior when
-        # this comment was written. This will not affect any real
-        # result record
-        if r:
-            yield r
-
-        # collect if summary is desired
-        if do_custom_result_summary:
-            results.append(r)
-
-    # result summary before a potential exception
-    # custom first
-    if do_custom_result_summary:
-        if pass_summary:
-            summary_args = (results, action_summary)
-        else:
-            summary_args = (results,)
-        interface.custom_result_summary_renderer(*summary_args)
-    elif result_renderer in ('generic', 'default') \
-            and action_summary \
-            and sum(sum(s.values())
-                    for s in action_summary.values()) > 1:
-        # give a summary in generic mode, when there was more than one
-        # action performed
-        render_action_summary(action_summary)
-
-    if incomplete_results:
-        raise IncompleteResultsError(
-            failed=incomplete_results,
-            msg="Command did not complete successfully")
-
-
-def _validate_cmd_call(interface: anInterface, kwargs: Dict) -> None:
-    """Validate a parameterization of a command call
-
-    This is called by `_execute_command_()` before a command call, with
-    the respective Interface sub-type of the command, and all its
-    arguments in keyword argument dict style. This dict also includes
-    the default values for any parameter that was not explicitly included
-    in the command call.
-
-    This expected behavior is to raise an exception whenever an invalid
-    parameterization is encountered.
-
-    This default implementation performs no validation.
-    """
-    pass
+    import warnings
+    from datalad.interface.base import eval_results as f
+    warnings.warn("datalad.interface.utils.eval_results is obsolete. "
+                  "Use datalad.interface.base.eval_results instead",
+                  DeprecationWarning)
+    return f(wrapped)
```

Comment:
minor:

```suggestion
    from datalad.interface.base import eval_results as eval_results_moved
    warnings.warn("datalad.interface.utils.eval_results is obsolete. "
                  "Use datalad.interface.base.eval_results instead",
                  DeprecationWarning)
    return eval_results_moved(wrapped)
```

to avoid confusing `f` in a possible traceback.